feeds/wlan-ap/opensync/files/bin/check_wan_link.sh

```diff
@@ -1,6 +1,21 @@
 #!/bin/sh
+. /usr/share/libubox/jshn.sh
 
-if="$(uci get network.wan.ifname)"
+json_init
+json_load "$(cat /etc/board.json)"
+json_select network
+	json_select "wan"
+		json_get_vars ifname
+	json_select ..
+json_select ..
+
+[ -n "$ifname" ] || {
+	ifname=$(uci get network.wan.ifname)
+	ifname=${ifname%% *}
+}
+
+
+if="$ifname"
```

Comment:
Have you tried using network_find_wan() from /lib/functions/network.sh?

Reply:
It uses ubus call network.interface dump internally, I don't see any info containing the ethX interfaces in it.